### PR TITLE
feat: node test runner

### PR DESCRIPTION
**Pull request by ilteoood**
Migrate the tests to use node test runner

### CODE REVIEW

File: test/integration.test.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-const { test } = require('tap')
+const { test } = require('node:test')
+
 const Fastify = require('fastify')
 
 const FastifySecrets = require('../lib/fastify-secrets-env.js')
 
 test('integration', async (t) => {
-  t.plan(1)
+  t.plan(2)
```
Comment (simoneb) [inline test/integration.test.js:10]: unnecessary

File: test/integration.test.js [hunk 2]
```diff
@@ -24,12 +25,6 @@ test('integration', async (t) => {
...
 
-  t.has(
-    fastify.secrets,
-    {
-      test1: 'test content 1',
-      test2: 'test content 2'
-    },
-    'decorates fastify with secret content'
-  )
+  t.assert.deepStrictEqual(fastify.secrets.test1, 'test content 1', 'decorates fastify with secret content')
```
Comment (simoneb) [inline test/integration.test.js:28]: why don't we assert in the same way as before?

File: test/client.test.js [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
     const client = new EnvClient()
     const secret = await client.get('SECRET')
 
-    t.equal(secret, 'secret content', 'gets data from env')
+    t.assert.deepStrictEqual(secret, 'secret content', 'gets data from env')
   })
 
-  t.test('sdk error', async (t) => {
+  test('sdk error', async (t) => {
     t.plan(1)
```
Comment (simoneb) [inline test/client.test.js:19]: let's remove these useless plans since we're there

File: test/integration.test.js (hunk 2 above)
Comment (ilteoood) [reply test/integration.test.js:28]: `has` is used to check for a subsect of the object, isn't it?
I think that the node test runner doesn't have this capability yet

**Review by simoneb — APPROVED** [review]

**Comment by ilteoood** [comment]
@simoneb thanks for the approval, but I cannot merge the PR by myself

**Pull request merged**